### DISCUSSION TITLE
Only expose target client's password with autologin

### DISCUSSION THREAD

**Discussion by alexkramer98** [ltsp/ltsp#810] [Q&A] [answered]
Hi everyone,

So we use a VMDK for the client image and a server for LTSP. 

Each client has autologin and its own user. Our current config looks something like this:
```
[clients]
HOSTNAME="ltsp-%{MAC}"
AUTOLOGIN="ltsp-%{MAC}"
RELOGIN=1

[ma:ca:dd:re:ss:01]
PASSWORDS_01="ltsp-%{MAC}/base64="

[ma:ca:dd:re:ss:02]
PASSWORDS_02="ltsp-%{MAC}/base64="
```
The user for each client is equal to the client's hostname. `base64=` will be a unique, randomly generated password and will be defined before the client gets introduced the first time. An entry will be present for each client.

However, we found out that all the client's passwords are present on every client, as `ltsp.conf` will be copied to them in full. Someone with malicious intent could just `cat /etc/ltsp/ltsp.conf` and get all (encoded) client passwords.

Our goal is for a client to *only* have write access to its own `/home/$hostname` folder on the server. 

Does anyone know how to approach this?

**Comment by alkisg** [comment, marked as answer]
The x86_64.img image and ltsp.conf contents are not considered private, even if their permission are "readable only by root". They can be read by anyone in the LAN that has read the LTSP code and has a bit of Linux knowledge.

To send secrets to clients before login, see an idea at https://github.com/ltsp/ltsp/issues/127#issuecomment-733511783
